internal/pkg/template/templates/environment/cf.yml

```diff
@@ -96,24 +96,60 @@ Resources:
           Value: disabled
           {{- end}}
 {{- end}}
-  PublicLoadBalancerSecurityGroup:
+  PublicHTTPLoadBalancerSecurityGroup:
```

Comment:
hmmm is this a safe change? 🤔 

Does renaming the logic id of a resource break environment updates? For example if I have already deployed an environment with `PublicLoadBalancerSecurityGroup` and now try `env deploy` with this new template does that succeed?

Reply:
After testing, yeah this does succeed at least in a basic circumstance.

Reply:
My concern is that it may delete the old security group and create a new one. If we run `copilot env deploy`  on an environment that still has `PublicLoadBalancerSecurityGroup` and not `PublicHTTPLoadBalancerSecurityGroup` yet (i.e. before your PR change), we'd probably see the following activities in `Event` tab:
```
PublicLoadBalancerSecurityGroup DELETE_COMPLETE
PublicLoadBalancerSecurityGroup DELETE_IN_PROGRESS
...
<stack_name> UPDATE_COMPLETE_CLEANUP_IN_PROGRESS
...
PublicHTTPLoadBalancerSecurityGroup CREATE_COMPLETE
PublicHTTPLoadBalancerSecurityGroup CREATE_IN_PROGRESS
...
<stack_name> UPDATE_IN_PROGRESS
```
Although it's probably not a problem to recreate the security group 🤔  but we should be careful about it if it introduces unnecessary resource updates just because the logical ID changes.

Reply:
Similarly, I'm also still worried because I don't have a clear understanding of the sequence of steps that occur.

> After testing, yeah this does succeed at least in a basic circumstance.

@CaptainCarpensir what did you observe on an env existing stack with a LBWS already enabled?
Does it:
1. Create the new `PublicHTTPLoadBalancerSecurityGroup` 
2. Update `PublicLoadBalancer`
3. Delete `PublicLoadBalancerSecurityGroup`

Is that the sequence? If it's the other way around (what Wanxian wrote) then the users will experience downtime because the load balancer for a brief period won't accept any incoming traffic.